django/db/migrations/autodetector.py

```diff
@@ -1187,14 +1188,41 @@ def create_altered_indexes(self):
 
             old_indexes = old_model_state.options[option_name]
             new_indexes = new_model_state.options[option_name]
-            add_idx = [idx for idx in new_indexes if idx not in old_indexes]
-            rem_idx = [idx for idx in old_indexes if idx not in new_indexes]
+            added_indexes = [idx for idx in new_indexes if idx not in old_indexes]
+            removed_indexes = [idx for idx in old_indexes if idx not in new_indexes]
+            renamed_indexes = []
+            # Find renamed indexes.
+            remove_from_added = []
+            remove_from_removed = []
+            for new_index in added_indexes:
+                new_index_dec = new_index.deconstruct()
+                new_index_name = new_index_dec[2].pop("name", None)
+                for old_index in removed_indexes:
+                    old_index_dec = old_index.deconstruct()
+                    old_index_name = old_index_dec[2].pop("name", None)
```

Comment:
The default argument for `pop` should not be necessary. When looking at the definition of  `db.models.indexes.Index.deconstruct`, the `name` is always defined

Reply:
Updated :+1: